packages/http-server/src/__tests__/server.oas.spec.ts

```diff
@@ -135,16 +123,58 @@ describe('server', () => {
     expect(response.statusCode).toBe(200);
   });
 
-  test('should return response even if there is no content defined in spec', async () => {
-    server = createServer({}, { components: {}, config: { mock: true } });
-    await server.prism.load({
-      path: resolve(__dirname, 'fixtures', 'no-responses.oas2.yaml'),
+  test('should validate the body params and return an error code', async () => {
+    const response = await server.fastify.inject({
+      method: 'POST',
+      url: '/pets',
+      payload: {
+        id: 1,
+        petId: 2,
+        quantity: 3,
+        shipDate: '12-01-2018',
+        status: 'placed',
+        complete: true,
+      },
     });
+    expect(response.statusCode).toBe(400);
+  });
 
-    const response = await server.fastify.inject({ method: 'GET', url: '/' });
+  if (file === 'petstore.oas2.json') {
```

Comment:
I cant figure out why we are splitting this with an if? What is v2 or v3 specific about thesew tests?

Reply:
This test in particular is testing something against an `oas2` file, and so it's oas2 specific.
The oas3 one is testing the `default` responses that apparently are not a thing in OAS2.

Does that clarify?

Reply:
`if`s in tests are evil :P 
why not simply move it outside the `each`?

Reply:
Yeah that's a good point, I'll move these outside. The reason I kept these inside is because I didn't want to reinit Prism again, but it's worth the duplication. Will fix!

Reply:
Done in 5e5d65f

Reply:
Good thing you moved them, but I still dont understand what is version specific about them. When I asked what is version specific sabout them you answered saying that they were version specific but... why. 🤣

Reply:
Ah sorry — it turns out default responses are a thing in OAS2 as well. Fixed in 27b2dc8